File: lib/provider/themeprovider.dart
import 'package:flutter/material.dart';

import '../model/apptheme.dart';
import '../model/contact.dart';

class Themeprovider extends ChangeNotifier {
  Apptheme apptheme = Apptheme(isdark: false);

  changetheme() {
    apptheme.isdark = !apptheme.isdark;
    notifyListeners();
  }

  favourite fav = favourite(isfav: false);
  List<Contact> allfav = [];

  favouriteadd(Contact contact) {
    fav.isfav = !fav.isfav;
    if (fav.isfav == true) {
      allfav.add(contact);
    }
    notifyListeners();
  }

  favouriteremove(Contact contact) {
    fav.isfav = !fav.isfav;
    if (fav.isfav == false) {
      allfav.remove(contact);
    }
    notifyListeners();
  }
}

File: lib/view/themeprovider.java
import 'package:counter/modal/apptheme.dart';
import 'package:flutter/material.dart';

class themeprovider extends ChangeNotifier {
  Apptheme apptheme = Apptheme(isdark: false);

  void changetheme(bool value) {
    apptheme.isdark = value;
    notifyListeners();
  }
}